enhancements/multi-arch/heterogeneous-architecture-clusters.md

```diff
@@ -0,0 +1,405 @@
+---
+title: heterogeneous-architecture-clusters
+authors:
+  - "@Prashanth684"
+reviewers:
+  - "@bparees"
+  - "@derekwaynecarr"
+  - "@jupierce"
+  - "@sdodson"
+  - "@wking"
+  - "@LalatenduMohanty - upgrades"
+  - "@sosiouxme - release payload build and release"
+  - "@enxebre - hypershift"
+  - "@dmage - imagestream api changes"
+  - "@kdoberst - console filtering changes"
+  - "@adambkaplan - Openshift Builds"
+  - "@perdasilva - OLM changes"
+  - "@mrunalp - MCO changes"
+  - "@JoelSpeed - Machine api changes"
+approvers:
+  - "@bparees"
+  - "@derekwaynecarr"
+api-approvers:
+  - "@JoelSpeed"
+creation-date: 2022-01-17
+last-updated: 2022-01-17
+tracking-link:
+  - https://issues.redhat.com/browse/RFE-2240
+  - https://issues.redhat.com/browse/OCPPLAN-4577
+status: implementable
+---
+
+# heterogeneous-architecture-cluster-support
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Operational readiness criteria is defined
+- [ ] Graduation criteria for tech preview and GA
+- [ ] User-facing documentation is created in [Openshift-docs](https://github.com/Openshift/Openshift-docs/)
+
+## Summary
+
+This enhancement describes the plan to support provisioning and upgrades of heterogeneous architecture Openshift clusters.
+A heterogeneous cluster is a cluster with nodes running on different architectures with a homogeneous control plane and worker machinesets
+supporting any combination of architectures (eg: x86 control plane + a mix of ARM/P/Z worker nodes). This document describes the support
+needed for heterogeneous architecture clusters in phases. There will be three major phases with basic support in phase 1 which is targeted
+for a 4.11 techpreview (with documented limitations) and the other phases which address gaps beyond the 4.11 time frame. The targeted platforms
+in the initial phases are Azure and AWS with support for x86 and ARM architectures.
+
+![alt heterogeneous cluster](heterogeneous-cluster.png)
+
+## Motivation
+
+- Users can take advantage of cost savings on architectures like ARM.
+- Users can colocate applications that require different architectures rather than needing multiple clusters and thus avoid cross cluster
+latency and connectivity failures.
+
+### Goals
+
+- Provide a way to install and upgrade clusters with varied architecture machinesets with the ability to autoscale.
```

Comment:
Provide a way to install and upgrade clusters with varied architecture Nodes with the ability to autoscale: In standalone OCP this is enabled through MachineSets, in Hypershift form factor via NodePools.

Reply:
updated